cli/internal/runcache/runcache.go

```diff
@@ -283,10 +284,10 @@ func (tc TaskCache) SaveOutputs(ctx context.Context, logger hclog.Logger, termin
 			terminal.Error(fmt.Sprintf("%s%s", ui.ERROR_PREFIX, color.RedString(" %v", fmt.Errorf("File path cannot be made relative: %w", err))))
 			continue
 		}
-		relativePaths[index] = relativePath
+		relativePaths[index] = fs.UnsafeToAnchoredSystemPath(relativePath)
 	}
 
-	if err = tc.rc.cache.Put(tc.pt.Pkg.Dir.ToStringDuringMigration(), tc.hash, duration, relativePaths); err != nil {
+	if err = tc.rc.cache.Put(tc.rc.repoRoot, tc.hash, duration, relativePaths); err != nil {
```

Comment:
This looks like a material change, but previously the value passed in here was unused.

It now uses this as the anchor for the `relativePaths` slice that is getting passed in here and prevents us from depending on `cwd`.